widgetastic_manageiq/__init__.py

```diff
@@ -94,6 +95,85 @@ def currently_selected(self):
             return None
 
 
+class SectionedBootstrapSelect(BootstrapSelect):
+    """ Refactor to allow for bootstrap select that have sections. """
+
+    SectionOption = namedtuple("SectionOption", ["text", "value", "section_id"])
+    SECTIONS = ".//li[contains(@class, 'dropdown-header')]"
+    OPTIONS = ".//li[contains(@class, 'data-original-index') and contains(@class, 'data-optgroup')]"
+    OPTION_ITEMS = "./div/ul/li"
+    SECTION_ELEMENTS = './/li[@data-optgroup="{}"]'
+    GROUP_INDEX = "data-optgroup"
+    ITEM_INDEX = "data-original-index"
+    ITEM_TEXT = ".//span[contains(@class, 'text')]"
+
+    @property
+    def all_sections(self):
+        return [
+            self.Option(self.browser.text(el), el.get_attribute(self.GROUP_INDEX))
+            for el in self.browser.elements(self.SECTIONS, parent=self)
+        ]
+
+    @property
+    def all_options(self):
+        return [
+            self.SectionOption(
+                self.browser.text(self.browser.element(self.ITEM_TEXT, parent=el)),
+                el.get_attribute(self.ITEM_INDEX),
```

Comment:
Nice, I feel like this is much more readable with the context of _what_ the locator is